From 0c147366da4fb829ba02a9db117e97a4107ac6a3 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Fri, 10 May 2024 11:31:44 +0100
Subject: [PATCH 1094/1135] dtoverlay: Add rgb-order override for
 vc4-kms-dpi-generic

Sets the new DT property for rgb-order.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/README                          | 4 ++++
 arch/arm/boot/dts/overlays/vc4-kms-dpi-generic-overlay.dts | 1 +
 2 files changed, 5 insertions(+)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -4988,6 +4988,10 @@ Params: clock-frequency         Display
                                 Set the default brightness. Normal range 1-16.
                                 (default 16).
         rotate                  Display rotation {0,90,180,270} (default 0)
+        rgb-order               Allow override of RGB order from DPI.
+                                Options for vc4 are "rgb", "bgr", "grb", and
+                                "brg". Other values will be ignored.
+
 
 
 Name:   vc4-kms-dpi-hyperpixel2r
--- a/arch/arm/boot/dts/overlays/vc4-kms-dpi-generic-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-dpi-generic-overlay.dts
@@ -77,5 +77,6 @@
 		rgb888 = <&panel_generic>, "bus-format:0=0x100a",
 			<&dpi_node_generic>, "pinctrl-0:0=",<&dpi_gpio0>;
 		bus-format = <&panel_generic>, "bus-format:0";
+		rgb-order = <&dpi_node_generic>, "rgb_order";
 	};
 };
